#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_missing_dataset_type_in_plugin.dpatch by Michael Prokop <mprokop@sipwise.com>
## based on http://git.tokkee.org/?p=collectd.git;a=commitdiff;h=fe3572bcfee8e3b612a2b8fd6a4aebef93233f9c
##
## DP: Collectd.pm: Added missing entry to the plugin (type id, type name) map.

@DPATCH@

diff --git a/bindings/perl/lib/Collectd.pm b/bindings/perl/lib/Collectd.pm
index f1b5d85..5a96ddd 100644
--- a/bindings/perl/lib/Collectd.pm
+++ b/bindings/perl/lib/Collectd.pm
@@ -116,6 +116,7 @@ my @fc_plugins : shared = ();
 my %cf_callbacks : shared = ();
 
 my %types = (
+	TYPE_DATASET,  "dataset",
 	TYPE_CONFIG,   "config",
 	TYPE_INIT,     "init",
 	TYPE_READ,     "read",
-- 
1.7.2.3

